src/sentry/sentry_metrics/indexer/indexer_consumer.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from typing import Any, Dict
+
+from django.conf import settings
+
+from sentry.sentry_metrics import indexer
+from sentry.sentry_metrics.indexer.base import UseCase
+from sentry.utils import json, kafka_config
+from sentry.utils.batching_kafka_consumer import AbstractBatchWorker
+from sentry.utils.kafka import create_batching_kafka_consumer
+from sentry.utils.pubsub import KafkaPublisher
+
+logger = logging.getLogger(__name__)
+
+snuba_metrics = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS]
+
+
+def get_metrics_consumer(topic=None, **options) -> None:
+    return create_batching_kafka_consumer(
+        {"ingest-metrics"}, worker=MetricsIndexerWorker(), **options
+    )
+
+
+class MetricsIndexerWorker(AbstractBatchWorker):
+    def process_message(self, message):
+        original: Dict[str, Any] = json.loads(message.value())
+        new_message = original.copy()
+        project_id = int(new_message["project_id"])
+
+        metric_id = indexer.resolve(project_id, UseCase.METRIC, new_message["name"])
```

Comment:
Shouldn't all calls to `indexer.resolve` go to `indexer.record` here? Because we want a new entry to be created when the metric / tag / value appears for the first time.

We would have to change the mock implementation of `record` to not throw when the name is one of the known strings, though.